framework/components/ADataTable/ADataTable.js

```diff
@@ -106,29 +121,63 @@ const ADataTable = forwardRef(
                             : "chevron-up"}
                         </AIcon>
                       )}
-                      {x.align === "end" ? x.name : ""}
-                    </th>
+                      {x.align === "end" && x.name}
+                    </TableHeader>
                   );
                 })}
-              </tr>
+                {ExpandableComponent && <TableHeader>Additional Details</TableHeader>}
+              </TableRow>
             </thead>
           )}
           <tbody>
-            {sortedItems.map((x, i) => (
-              <tr key={`a-data-table_row_${i}`}>
-                {headers.map((y, j) => (
-                  <td
-                    key={`a-data-table_cell_${j}`}
-                    className={`text-${y.align || "start"} ${
-                      y.cell?.className || ""
-                    }`.trim()}>
-                    {y.cell && y.cell.component
-                      ? y.cell.component(x)
-                      : x[y.key]}
-                  </td>
-                ))}
-              </tr>
-            ))}
+            {sortedItems.map((x, i) => {
+              const id = `a-data-table_row_${i}`;
+              const hasExpandedRowContent = ExpandableComponent &&
+                (typeof expandable.isRowExpandable === 'function'
+                  ? expandable.isRowExpandable(x)
+                  : true
+                );
+              return (
+                <TableRow data-expandable-row={hasExpandedRowContent} key={id}>
```

Comment:
Re: `key={id}` when `const id = 'a-data-table_row_${i}'`

When using a key based on the row's numeric index, sorting large tables will be slower. If you can use something unique to each row here (or a generated UUID, as a last resort), React can reconcile the DOM updates faster.

This might not be noticeable until you hit really complex use cases.

Reply:
I currently just went with an approach similar to the one used in [`useAToaster`](https://github.com/cisco-sbg-ui/atomic-react/blob/master/framework/components/AToaster/useAToaster.js) as to not bring in another library right now. The good news it that @frattaro brought up that [React version 18 is planning to ship a `useId` hook for improving hydration updates](https://github.com/reactwg/react-18/discussions/111).